ext/standard/user_filters.c

```diff
@@ -369,38 +372,42 @@ PHP_FUNCTION(stream_bucket_make_writeable)
 static void php_stream_bucket_attach(int append, INTERNAL_FUNCTION_PARAMETERS)
 {
 	zval *zbrigade, *zobject;
-	zval *pzbucket, *pzdata;
+	zval *pzbucket, *pzdata, rv;
 	php_stream_bucket_brigade *brigade;
 	php_stream_bucket *bucket;
 
 	ZEND_PARSE_PARAMETERS_START(2, 2)
 		Z_PARAM_RESOURCE(zbrigade)
-		Z_PARAM_OBJECT(zobject)
+		Z_PARAM_OBJECT_OF_CLASS(zobject, stream_bucket_class_entry)
 	ZEND_PARSE_PARAMETERS_END();
 
-	if (NULL == (pzbucket = zend_hash_str_find_deref(Z_OBJPROP_P(zobject), "bucket", sizeof("bucket")-1))) {
-		zend_argument_value_error(2, "must be an object that has a \"bucket\" property");
+	if ((brigade = (php_stream_bucket_brigade*)zend_fetch_resource(
+		Z_RES_P(zbrigade), PHP_STREAM_BRIGADE_RES_NAME, le_bucket_brigade)) == NULL) {
 		RETURN_THROWS();
 	}
 
-	if ((brigade = (php_stream_bucket_brigade*)zend_fetch_resource(
-					Z_RES_P(zbrigade), PHP_STREAM_BRIGADE_RES_NAME, le_bucket_brigade)) == NULL) {
+	if (NULL == (pzbucket = zend_read_property(NULL, Z_OBJ_P(zobject), "bucket", sizeof("bucket")-1, false, &rv))) {
+		zend_argument_value_error(2, "must be an object that has a \"bucket\" property");
 		RETURN_THROWS();
 	}
+	ZVAL_DEREF(pzbucket);
 
 	if ((bucket = (php_stream_bucket *)zend_fetch_resource_ex(pzbucket, PHP_STREAM_BUCKET_RES_NAME, le_bucket)) == NULL) {
 		RETURN_THROWS();
 	}
 
-	if (NULL != (pzdata = zend_hash_str_find_deref(Z_OBJPROP_P(zobject), "data", sizeof("data")-1)) && Z_TYPE_P(pzdata) == IS_STRING) {
-		if (!bucket->own_buf) {
-			bucket = php_stream_bucket_make_writeable(bucket);
-		}
-		if (bucket->buflen != Z_STRLEN_P(pzdata)) {
-			bucket->buf = perealloc(bucket->buf, Z_STRLEN_P(pzdata), bucket->is_persistent);
-			bucket->buflen = Z_STRLEN_P(pzdata);
+	if (NULL != (pzdata = zend_read_property(NULL, Z_OBJ_P(zobject), "data", sizeof("data")-1, false, &rv))) {
+		ZVAL_DEREF(pzdata);
+		if (Z_TYPE_P(pzdata) == IS_STRING) {
```

Comment:
Can it be anything else than string? And does it actually need deref?

Reply:
It's probably better to have that check just in case but was just wondering if it's even possible...

Reply:
Ok thought that with read only it will be always non ref string but haven't looked properly. It's ok to keep it in any case.